engines/myst3/myst3.cpp

```diff
@@ -1505,28 +1512,43 @@ void Myst3Engine::tryAutoSaving() {
 	_lastSaveTime = _system->getMillis();
 
 	// Get a thumbnail of the game screen
-	Common::DisposablePtr<Graphics::Surface, Graphics::SurfaceDeleter> thumbnail(nullptr, DisposeAfterUse::NO);
-	if (_menu->isOpen()) {
-		thumbnail.reset(_menu->borrowSaveThumbnail(), DisposeAfterUse::NO);
-	} else {
-		// Capture thumbnail before saving
-		thumbnail.reset(_menu->captureThumbnail(), DisposeAfterUse::YES);
-	}
+	if (!_menu->isOpen())
+		_menu->generateSaveThumbnail();
 
 	// Open the save file
 	Common::String saveName = "Autosave";
-	Common::String fileName = Saves::buildName(saveName.c_str(), getPlatform());
-	Common::OutSaveFile *save = getSaveFileManager()->openForSaving(fileName);
-	if (!save) {
-		warning("Unable to open the autosave file: %s.", fileName.c_str());
-	}
+	saveGame(saveName);
 
-	// Save the state and the thumbnail
-	if (!_state->save(save, saveName, thumbnail.get())) {
-		warning("Unable to write the autosave to '%s'.", fileName.c_str());
+}
+
+void Myst3Engine::saveGame(Common::String &saveName) {
```

Comment:
Is there any reason why you're not implementing `Common::Error saveGameState(int slot, const Common::String &desc)` and adding `kSupportsSavingDuringRuntime` to the features list so saving actually works from the GMM?

Reply:
The game crashes if I try to save without a name (using the auto generated descriptor based on timestamps). 

Should I write my own MainMenuDialog::save()?

residualvm: backends/fs/posix/posix-fs.cpp:113: virtual AbstractFSNode* POSIXFilesystemNode::getChild(const Common::String&) const: Assertion `!n.contains('/')' failed.
Aborted (core dumped)

Reply:
I think it crashes because the filename has the following form when using the default save description: `%04d-%02d-%02d / %02d:%02d:%02d`.  The slash is not allowed in filenames.

IMO the engine should replace invalid characters with something harmless in `saveGameState` or disallow save names that contain invalid characters like the Pegasus engine in ScummVM: https://github.com/scummvm/scummvm/blob/4007a7088a1c4266abfe466540a70711b9050926/engines/pegasus/pegasus.cpp#L715